src/trading/appDataUtils.ts

```diff
@@ -34,3 +39,29 @@ export async function generateAppDataFromDoc(
 
   return { fullAppData, appDataKeccak256 }
 }
+
+export async function mergeAppDataDoc(
+  _doc: LatestAppDataDocVersion,
+  appDataOverride: AppDataParams
+): Promise<AppDataInfo> {
+  // Do not merge hooks if there are overrides
+  // Otherwise we will just append hooks instead of overriding
+  const doc = appDataOverride.metadata?.hooks
+    ? {
+        ..._doc,
+        metadata: {
+          ..._doc.metadata,
+          hooks: {},
```

Comment:
But here hooks are not being merged nor appended, they are reset to empty. No?

Reply:
They are merged bellow:
```
const appData = (appDataOverride ? deepmerge(doc, appDataOverride) : doc) as LatestAppDataDocVersion
```